compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -3425,14 +3429,9 @@ impl<'a> Parser<'a> {
         let if_span = self.prev_token.span;
         let mut cond = self.parse_match_guard_condition()?;
 
-        CondChecker::new(self).visit_expr(&mut cond);
+        CondChecker::new(self, LetChainsPolicy::AlwaysAllowed).visit_expr(&mut cond);
```

Comment:
Would be nice to leave a note here that this is allowed since it is already gated behind `if_let_guard`, so it is still unstable and not being stabilized. But that can be done as a followup.